ouroboros-consensus/test/consensus-test/Test/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -829,6 +829,13 @@ instance Arbitrary ChainSyncClientSetup where
     , y <- shrink x
     ]
 
+chooseExponential :: Enum a => Int -> (a, a) -> Gen a
+chooseExponential decayFactor (low, up) =
+  frequency
+    [ (1, pure low)
+    , (1, chooseExponential decayFactor (succ low, up))
+    ]
```

Comment:
`decayFactor` is completely unused by this function, it is just passed around recursively. I imagine it is supposed to influence the weights in some way?

Also: what about moving this to `Test.Util.QuickCheck` as well as adding documentation? Maybe take inspiration from [hedgehog](https://hackage.haskell.org/package/hedgehog-1.5/docs/Hedgehog-Range.html#g:5)?

Reply:
oops, was faffing around with `decayFactor` and accidentally committed it while it did nothing lmao

I don't really think this a great function, and it probably shouldn't be reused elsewhere? `hedgehog`'s `Range.exponential` is exactly what we want here, but I don't want to badly reimplement a bunch of `Hedgehog.Range` in our codebase 😅

Reply:
Makes sense, then let's keep this as an ad-hoc helper here for now :+1: